src/VisualStudio/Core/Def/SolutionBuildMonitor.cs

```diff
@@ -22,6 +22,7 @@ public SolutionBuildMonitor(VisualStudioWorkspace workspace)
             {
                 _notificationService = notificationService;
                 KnownUIContexts.SolutionBuildingContext.UIContextChanged += SolutionBuildingContextChanged;
+                KnownUIContexts.SolutionOpeningContext.UIContextChanged += SolutionOpeningContext_UIContextChanged;
```

Comment:
you probably want to change name of this file. to like SolutionContextMonitoring.